ls1intum/Artemis
# [Enhancement] Only fetch build logs if the build failed

### Checklist

### Motivation and Context
Artemis (client) currently fetches build logs if it doesn't receive any feedback. This is a problem for exercises with hidden tests, as the build logs leak hidden test feedback.

### Description
_Client:_
- Only fetch build logs if we have a result and the submission is marked as failed. If no submission is available in the result object, attempt to fetch the build logs.
- Consolidate to one service for fetching of build logs
- Filter out static code analysis feedback before sending feedback to Orion
- Provide separate services for each code editor instance in the exam
- Add/Fix tests

_Server:_
- Fetch submission but don't fetch feedback (not used) for retrieval of build logs
- Return forbidden if `submission.buildFailed` is false
- Add integration tests

### Steps for Testing


_Exam_
1. Create an exam with multiple programming exercises
2. Participate in the exam
3. Create, delete, edit files and save and commit them or refresh files
--> Code editors should stay in a consistent state and not mix up requests (can check this also in the network tab)
--> Check that instructions and UML diagrams are rendered independently for each exercise
4. Provoke build errors
--> Build logs should been shown for the actual exercise

_Exercise with hidden tests_
1. Create an programming exercise with tests after due date
2. Navigate to test case management and set all test cases to run after due date
3. Participate in the exercise
4. Submit a solution which builds
--> Build logs should not be shown
5. Submit a solution which does not compile
--> Build logs should be shown
